### PR TITLE
Add the Breadcrumbs component

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Create the Breadcrumbs component

#### What problem is this solving?
Give to the user the ability to go back to a previous point

#### How should this be manually tested?
[workspace](https://felipe--storecomponents.myvtex.com/smartphones/s)

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,42 @@
...
+$RECYCLE.BIN/
+
+# App specific
+node_modules/
+docs/_book/
+.tmp
+.idea
+npm-debug.log
+.build/
+lib
```
Comment (brunojdo) [inline .gitignore]: `*.orig`

Comment (brunojdo) [reply .gitignore]: `package-lock.json`

File: README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+| ------------------ | -------------- | --------------------------------------------------------------------------- |
+| `search`           | `String`       | Term used to get to that specific page on the website                       |
+| `slug`             | `String`       | Product's unique identification stored on the database                      |
+| `categories`       | `Array(String)`| List of categories which the product belongs to                             |
+
+## Extra info:
+
+The product's categories should appear as an array in one of this two formats:
+
+1 -
```
Comment (lucasecdb) [inline README.md]: ⛏ `- 1`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Breadcrumb
+Breadcrumb is a canonical component that any VTEX app can use.
+
+To use it in your code you should first add it as a dependencie into your app's ```manifest.json```, like:
+
+```json
+"dependencies": {
+    "vtex.breadcrumb": "0.x",
+    "vtex.category-menu": "0.x"
```
Comment (brunojdo) [inline README.md]: remove this

File: README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+}
+```
+
+## Usage
+You can use it in your code through an extension point in your app.
+
+```javascript
+    <ExtensionPoint
```
Comment (lucasecdb) [inline README.md]: Show direct usage of the component, not the `ExtensionPoint`

Comment (salesfelipe) [reply README.md]: But the component will never be used directly, whats the point of that?

File: README.md (hunk 3 above)
Comment (salesfelipe) [reply README.md]: the category-menu or the whole dependencies ... ?

File: .gitignore (hunk 1 above)
Comment (salesfelipe) [reply .gitignore]: 👍

File: README.md (hunk 2 above)
Comment (salesfelipe) [reply README.md]: 👍

File: README.md (hunk 3 above)
Comment (brunojdo) [reply README.md]: category-menu

File: README.md (hunk 4 above)
Comment (lucasecdb) [reply README.md]: The point is to document the usage of the _own_ component, so in the future, if we remove it from the extension point and use it directly or somebody else need to use it in another app but don't want to put in an extension point, we'll have the proper documentation for that

File: README.md (hunk 3 above)
Comment (salesfelipe) [reply README.md]: 👍

File: README.md (hunk 4 above)
Comment (salesfelipe) [reply README.md]: Even though I agree with you in parts, I strongly believe that the document should reflect the real world usage, not a theoretical version of it. Moreover, this document is not immutable, so if we change the implementation of the component or just the usage of it, this changes should be reflected in this document. Therefore, I ask you to reconsider your change request.

**Pull request merged**